classes/battle/BattleManager.php

```diff
@@ -1561,4 +1560,8 @@ protected function chooseAndSetAIAction(Fighter $ai) {
     public function getPatrolId(): int {
         return $this->battle->patrol_id;
     }
+
+    public function getEffects(): BattleEffectsManager {
+        return $this->effects;
+    }
```

Comment:
```suggestion
    /**
      * @return BattleEffect[]
      */
    public function getActiveEffects(): array {
        return $this->effects->active_effects;
    }
```
If possible let's do this instead so the full options of the effects manager are not exposed (this is to force effects logic into this class or an explicit dependency)

Reply:
Makes sense.